docs/KUBERNETES.md

```diff
@@ -187,6 +187,27 @@ Alternatively, update the DaemonSet configuration directly by using the followin
 kubectl patch daemonset -n logdna-agent logdna-agent --type json -p '[{"op":"add","path":"/spec/template/spec/containers/0/securityContext/runAsUser","value":5000},{"op":"add","path":"/spec/template/spec/containers/0/securityContext/runAsGroup","value":5000}]'
 ```
 
+### Enabling persistent agent state
+
+To avoid skipping or duplicating logs when the agent pod restarts or is replaced the agent needs to store it's progress on the host node's filesystem. This is achieved using a hostPath volume. The host directory must be writable by the user or group specified in the securityContext.
+
+To achieve this the host directory must either already exist with the correct
+permissions or else Kubernetes will create the directory without write permissions for the agent user.
+
+In this case the permissions must be set before the agent starts. When running as non-root the agent pod does not have permissions to do this, so an initcontainer may be used.
+
```

Comment:
Suggested rewrite of lines 194-197: "To achieve this the host directory must already exist with the correct permissions before the agent starts (otherwise Kubernetes will create the directory without the needed write permissions for the agent user). When running as non-root the agent pod does not have permissions to configure the host directory, so we use an *init container*, which is run before the app containers are started, to set the host directory permissions."

Reply:
Not sure why "init container" would be italicized above. We'd need to do the same in other places in the doc. Also, I don't believe *we* use an init container - we're telling the user to use one

Reply:
the use of italics is standard for the first time a new term is introduced. A way to call attention to it, make sure the reader sees it. So we would only use it once. BUt.. we can also drop it, or use `init container`.

Reply:
Got that--but do we do the same for first use of other terms in the doc? That was my concern.